src/github.rs

```diff
@@ -48,7 +48,7 @@ async fn upload_release_artifact(
     dry_run: bool,
 ) -> Result<()> {
```

Comment:
The code changes in this file are just making `upload_release_artifact` more robust in dry-run, such that if the release itself doesn't exist, it can still do some validation.